src/Models/Media.php

```diff
@@ -165,10 +168,73 @@ public function search($query)
         return $this->client->request('POST', 'media/search', compact('query'));
     }
 
+    /**
+     * Upload a media file.
+     *
+     * @param string $path
+     *
+     * @return array
+     */
+    public function upload($path)
+    {
+        $file = fopen($path, 'r');
+        $stream = Psr7\stream_for($file);
+
+        $fileName = basename($path);
+        $mimeType = mime_content_type($file);
+
+        $vaporParams = ['content_type' => $mimeType];
+
+        $signedUrl = $this->client->request('POST', 'vapor/signed-storage-url', $vaporParams);
+
+        if (is_array($signedUrl) && $signedUrl['url']) {
+            $headers = $signedUrl['headers'];
+            unset($headers['Host']);
+
+            $this->uploadToS3($signedUrl, $headers, $fileName, $stream);
+
+            $uploadData = ['key' => $signedUrl['key'], 'file_name' => $fileName];
+
+            $response = $this->client->request('POST', 'upload', $uploadData);
+        }
+
+        return $response;
+    }
+
+    /**
+     * Upload the file stream to s3.
+     *
+     * @param array $signedUrl
+     * @param array $headers
+     * @param string $fileName
+     * @param GuzzleHttp\Psr7 $stream
+     *
+     * @return void
+     */
+    protected function uploadToS3($signedUrl, $headers, $fileName, $stream)
+    {
+        $client = new GuzzleClient();
+        $request = new Request(
+            'PUT',
+            $signedUrl['url'],
+            ['headers' => json_encode($headers)],
+            new Psr7\MultipartStream(
+                [
+                    [
+                        'name' => $fileName,
+                        'contents' => $stream,
+                    ],
+                ]
+            )
+        );
+
+        $client->send($request);
+    }
+
     /**
      * Archive a media.
      *
-     * @param int $id
+     * @param array $id
```

Comment:
We should rename this var to `$ids` but at the same time we should also accept an int-value for just logical reasons and to cover this edge-case in case the user just enters an id.